src/wui/shipwindow.cc

```diff
@@ -649,7 +649,7 @@ void ShipWindow::act_construct_port() {
 		return;
 	}
 	if (Widelands::Game* game = ibase_.get_game()) {
-		game->send_player_ship_construct_port(*ship, ship->exp_port_spaces().front());
+		game->send_player_ship_construct_port(*ship, ship->exp_port_spaces().back());
```

Comment:
Instead of exchanging front→back everywhere with the risk of overlooking something next time we change something, it might be better to add a new function to Ship like
```c++
[[nodiscard]] const Coords& current_port_space() const {
	return exp_port_spaces().back();
}
```

Reply:
Relatedly, there is now a mismatch between which portspace is visually indicated as the current one versus which one can currently be attacked by a warship

Reply:
> Instead of exchanging front→back everywhere with the risk of overlooking something next time we change something, it might be better to add a new function to Ship like

I did that first, but undid it because now InteractivePlayer keeps and uses a cache of the whole vector, and takes the primary element from that. So one major user is left out already. I'll try to solve it.

> there is now a mismatch between which portspace is visually indicated as the current one versus which one can currently be attacked by a warship

That's strange, I didn't notice that. Will test it.

Reply:
> > there is now a mismatch between which portspace is visually indicated as the current one versus which one can currently be attacked by a warship
> 
> That's strange, I didn't notice that. Will test it.

Confirmed. Sorry about it, will try to fix too.

Reply:
> there is now a mismatch between which portspace is visually indicated as the current one versus which one can currently be attacked by a warship

That was actually just a use of `.front()` on another missed ref. No longer possible (outside of `Ship`), the list isn't exposed any more.